pkg/collector/git_collector.go

```diff
@@ -162,8 +165,15 @@ func (c *GitCollector) generateURL(storageOrg string, storageRepoName string, rP
 	if !c.gitInfo.IsGitHub() && gits.SaasGitKind(c.gitInfo.Host) == gits.KindGitHub {
 		url = fmt.Sprintf("https://raw.%s/%s/%s/%s/%s", c.gitInfo.Host, storageOrg, storageRepoName, c.gitBranch, rPath)
 	} else {
-		// TODO only supporting github for now!!!
-		url = fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/%s/%s", storageOrg, storageRepoName, c.gitBranch, rPath)
+		switch c.gitKind {
+		case gits.KindGitlab:
+			url = fmt.Sprintf("https://%s/%s/%s/-/raw/%s/%s", c.gitInfo.Host, storageOrg, storageRepoName, c.gitBranch, rPath)
+		case gits.KindBitBucketServer:
+			url = fmt.Sprintf("https://%s/projects/%s/repos/%s/raw/%s?at=%s", c.gitInfo.Host, storageOrg, storageRepoName, rPath, neturl.QueryEscape(fmt.Sprintf("refs/heads/%s", c.gitBranch)))
+		default:
+			// Fall back on GitHub for now
+			url = fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/%s/%s", storageOrg, storageRepoName, c.gitBranch, rPath)
```

Comment:
What happens if the `gitKind` is not `GitHub`? 

Would this code not break?

Reply:
Well, yeah, but it's already broken. =) We handle the GitHub Enterprise case in the first part of this, before the `switch` I've added, and until now, the only alternative was this line. With the `switch`, we at least check GitLab and BitBucket Server cases and construct the appropriate URLs for those.